core/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -303,14 +306,16 @@ public Builder version(Integer version) {
         }
 
         public Builder template(String template) {
-            this.template = template;
+            this.template = Collections.singletonList(template);
```

Comment:
I'd probably just convert the call sites rather than add a new method like this.